core/src/main/java/io/grpc/internal/ServerImpl.java

```diff
@@ -179,13 +181,18 @@ public ServerImpl shutdown() {
     return this;
   }
 
-  /**
-   * NOT YET IMPLEMENTED. This method currently behaves identically to shutdown().
-   */
-  // TODO(ejona86): cancel preexisting calls.
   @Override
   public ServerImpl shutdownNow() {
     shutdown();
+    Collection<ServerTransport> transportsCopy;
+    Status nowStatus = Status.UNAVAILABLE.withDescription("Server shutdownNow invoked");
+    synchronized (lock) {
+      shutdownNowStatus = nowStatus;
+      transportsCopy = new ArrayList<ServerTransport>(transports);
+    }
+    for (ServerTransport transport : transportsCopy) {
+      transport.shutdownNow(nowStatus);
```

Comment:
This doesn't seem required for correctness. Are you doing this just to shutdownNow() the transports as the earliest possible time?